addons/stock/models/stock_move.py

```diff
@@ -415,16 +415,22 @@ def _compute_forecast_information(self):
         not_product_moves = self.filtered(lambda move: move.product_id.type != 'product')
         for move in not_product_moves:
             move.forecast_availability = move.product_qty
+
+        product_moves = (self - not_product_moves)
+        warehouse_by_location = {loc: loc.get_warehouse() for loc in product_moves.location_id}
+
         outgoing_unreserved_moves_per_warehouse = defaultdict(lambda: self.env['stock.move'])
-        for move in (self - not_product_moves):
+        for move in product_moves:
             picking_type = move.picking_type_id or move.picking_id.picking_type_id
             is_unreserved = move.state in ('waiting', 'confirmed', 'partially_available')
             if picking_type.code in self._consuming_picking_types() and is_unreserved:
-                outgoing_unreserved_moves_per_warehouse[picking_type.warehouse_id] |= move
+                outgoing_unreserved_moves_per_warehouse[warehouse_by_location[move.location_id]] |= move
             elif picking_type.code in self._consuming_picking_types():
                 move.forecast_availability = move.reserved_availability
 
         for warehouse, moves in outgoing_unreserved_moves_per_warehouse.items():
+            if not warehouse.view_location_id.id:  # location doesn't belong to a warehouse "Not available" (no prediction)
```

Comment:
Not a big fan of this comment, I would prefer something simplier (say if I'm wrong):
```suggestion
            if not warehouse.view_location_id:  # No prediction possible if no warehouse.
```
Also, check `view_location_id` is enough I think, no need to look the `id`.